src/Databases/DatabaseOnDisk.cpp

```diff
@@ -193,25 +194,10 @@ void DatabaseOnDisk::createTable(
     if (!create.attach)
         checkMetadataFilenameAvailability(table_name);
 
-    if (create.attach && fs::exists(table_metadata_path))
-    {
-        ASTPtr ast_detached = parseQueryFromMetadata(log, local_context, table_metadata_path);
```

Comment:
Why do you think that it's unreachable?

Reply:
I tried to construct several test cases. None of them can reach this code. I'm not 100% confident though.

Reply:
There will be an exception without this code. Something like `File /data/store/03c/03c816c3-f74b-4230-a7fa-6693a5bb8b72/t.sql exists`.

If the purpose of this code is for better exception message, we need to fix the message as it's not right:

```
Code: 57. DB::Exception: Received from localhost:9000. DB::Exception: Table `default`.`t` already exist (detached permanently). To attach it back you need to use short ATTACH syntax or a full statement with the same UUID. (TABLE_ALREADY_EXISTS)
```

We cannot attach it back with the same UUID. It will lead to `Mapping for table with UUID=4640358e-920e-49d1-ab00-a68e6e3229af already exists. It happened due to UUID collision, most likely because some not random UUIDs were manually specified in CREATE queries`

Reply:
#39315